include/deal.II/cgal/polygon.h

```diff
@@ -0,0 +1,115 @@
+// ------------------------------------------------------------------------
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2025 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// Part of the source code is dual licensed under Apache-2.0 WITH
+// LLVM-exception OR LGPL-2.1-or-later. Detailed license information
+// governing the source code and code contributions can be found in
+// LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
+//
+// ------------------------------------------------------------------------
+
+#ifndef dealii_cgal_polygon_h
+#define dealii_cgal_polygon_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_CGAL
+#  include <deal.II/cgal/point_conversion.h>
+#  include <deal.II/cgal/utilities.h>
+
+#  include <CGAL/Polygon_2.h>
+#  include <CGAL/Polygon_with_holes_2.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace CGALWrappers
+{
+  /**
+   * Build a CGAL::Polygon_2 from a deal.II cell.
+   *
+   * The class Polygon_2 is a wrapper around a container of points that can
+   * be used to represent polygons.
+   * The points must be added in counterclockwise order to a Polygon_2
+   *
+   * More information on this class is available at
+   * https://doc.cgal.org/latest/Polygon/index.html
+   *
+   * The functions are for two dimensional triangulations in two dimensional
+   * space. Projecting 3D points is possible with CGAL but not implemented.
+   *
+   * The generated boundary representation is useful when performing
+   * geometric operations using compute boolean operations.
```

Comment:
Did you mean "set boolean operations"? I can't quite understand this sentence :-)